boards/entrypoint.sh

```diff
@@ -0,0 +1,115 @@
+set -e 
+
+if [ -z "$AZURE_BOARDS_ORGANIZATION" ]; then
+    echo "AZURE_BOARDS_ORGANIZATION is not set." >&2
+    exit 1
+fi
+
+if [ -z "$AZURE_BOARDS_PROJECT" ]; then
+    echo "AZURE_BOARDS_PROJECT is not set." >&2
+    exit 1
+fi
+
+if [ -z "$AZURE_BOARDS_TOKEN" ]; then
+    echo "AZURE_BOARDS_TOKEN is not set." >&2
+    exit 1
+fi
+
+if [ -z "$GITHUB_EVENT_PATH" ]; then
+    echo "GITHUB_EVENT_PATH is not set." >&2
+    exit 1
+fi
+
+function parse_markdown {
+    markdown -f fencedcode -f githubtags
+}
+
+function create_work_item {
+    echo "Creating work item..."
+    HYPERLINK="Created from <a href='${GITHUB_ISSUE_HTML_URL}'>Issue #${GITHUB_ISSUE_NUMBER}</a>"
+    RESULTS=$(vsts work item create --type "${AZURE_BOARDS_TYPE}" \
+        --title "${AZURE_BOARDS_TITLE}" \
+        --description "${AZURE_BOARDS_DESCRIPTION}" \
+        -f System.Tags="GitHub; Issue ${GITHUB_ISSUE_NUMBER}" \
+        --discussion "${HYPERLINK}" \
+        --output json)
+    AZURE_BOARDS_ID=$(echo "${RESULTS}" | jq --raw-output .id)
+
+    echo "Created work item #${AZURE_BOARDS_ID}"
+}
+
+function work_items_for_issue {
+    vsts work item query --wiql "SELECT ID FROM workitems WHERE [System.Tags] CONTAINS 'GitHub' AND [System.Tags] CONTAINS 'Issue ${GITHUB_ISSUE_NUMBER}'" | jq '.[].id' | xargs
+}
+
+AZURE_BOARDS_TYPE="${AZURE_BOARDS_TYPE:-Feature}"
+AZURE_BOARDS_CLOSED_STATE="${AZURE_BOARDS_CLOSED_STATE:-Closed}"
+AZURE_BOARDS_REOPENED_STATE="${AZURE_BOARDS_REOPENED_STATE:-New}"
+AZURE_DEVOPS_URL="https://dev.azure.com/${AZURE_BOARDS_ORGANIZATION}/"
+
+vsts configure --defaults instance="${AZURE_DEVOPS_URL}" project="${AZURE_BOARDS_PROJECT}"
+
+vsts login --token "${AZURE_BOARDS_TOKEN}"
+
+GITHUB_EVENT=$(jq --raw-output 'if .comment != null then "comment" else empty end' "$GITHUB_EVENT_PATH")
+GITHUB_EVENT=${GITHUB_EVENT:-$(jq --raw-output 'if .issue != null then "issue" else empty end' "$GITHUB_EVENT_PATH")}
+GITHUB_ACTION=$(jq --raw-output .action "$GITHUB_EVENT_PATH")
+GITHUB_ISSUE_NUMBER=$(jq --raw-output .issue.number "$GITHUB_EVENT_PATH")
+GITHUB_ISSUE_HTML_URL=$(jq --raw-output .issue.html_url "$GITHUB_EVENT_PATH")
+AZURE_BOARDS_TITLE=$(jq --raw-output .issue.title "$GITHUB_EVENT_PATH")
+AZURE_BOARDS_DESCRIPTION=$(jq --raw-output .issue.body "$GITHUB_EVENT_PATH" | parse_markdown)
+
+TRIGGER="${GITHUB_EVENT}/${GITHUB_ACTION}"
+
+case "$TRIGGER" in
+"issue/opened")
+    # If there's a GitHub issue label configured then don't create a
+    # corresponding Azure Boards work item.  Wait for a labelled event
+    # to create the work item.
+    if [ -z "$ISSUE_LABEL" ]; then
+        create_work_item
+    fi
+    ;;
+
+"issue/labeled")
+    # If there's a GitHub issue label configured then see if that was the
+    # label applied.  If so, create a new Azure Boards work item to
+    # correspond to this issue.
+    NEW_LABEL=$(jq --raw-output .label.name "$GITHUB_EVENT_PATH")
+
+    if [ ! -z "$ISSUE_LABEL" ] && [[ "$NEW_LABEL" == "$ISSUE_LABEL" ]]; then
```

Comment:
Conver `! -z` to `-n`

Reply:
@ethomson, thoughts?

Reply:
I think `! -z` is much more readable, but I also like my bikesheds to be blue, so 🤷‍♂️

Reply:
Hahah! @mmitrik this would cause a `shellcheck` failure. FYI: [SC2236](https://github.com/koalaman/shellcheck/wiki/SC2236)

Reply:
Fixed.

Reply:
Good call, that settles it then.  Thanks @ds-ms